lib/Support/Scheduling/ASAPScheduler.cpp

```diff
@@ -0,0 +1,70 @@
+//===- ASAPScheduler.cpp - As-soon-as-possible list scheduler ---*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements an acyclic, as-soon-as-possible list scheduler.
+//
+//===----------------------------------------------------------------------===//
+#include "circt/Support/Scheduling/ASAPScheduler.h"
+
+#include <list>
```

Comment:
Super nit pick, but I'd recommend using std::deque for this.

Reply:
Fixed, thanks!